src/components/Map.js

```diff
@@ -336,6 +358,313 @@ class MapComponent extends Component {
+    
+    https://maps1.project.lizard.net/
+    geoserver/s0175_ijgenzon/wms?
+    REQUEST=GetFeatureInfo&
+    SERVICE=WMS&
+    SRS=EPSG%3A4326&
+    VERSION=1.1.1&
+    INFO_FORMAT=application%2Fjson&
+    BBOX=5.84026336669922%2C50.89870240470785%2C6.110801696777345%2C51.02844005602989&
+    HEIGHT=600&
+    WIDTH=788&
+    LAYERS=r0262_resultaten_workshopronde2_oplossingsrichtingen_20180228&
+    QUERY_LAYERS=r0262_resultaten_workshopronde2_oplossingsrichtingen_20180228&
+    FEATURE_COUNT=100
+    &X=315
+    &Y=422
+    //*/
+
+    return (
+      layerUrl +
+      L.Util.getParamString(
+        {
+          request: "GetFeatureInfo",
+          service: "WMS",
+          // // Why does this only seem to yield results with EPSG:4326 ?
+          // // srs of layer specifies it otherwise..
+          // srs: srs,
+          srs: "EPSG:4326",
+          version: "1.1.1", // I don't get results with 1.3.0 for some reason
+          info_format: "application/json",
+          bbox: bbox,
+          height: size.y,
+          width: size.x,
+          layers: layerName,
+          query_layers: layerName,
+          // Return multiple features of a layer if the layer has multiple
+          // features on the same location.
+          // http://docs.geoserver.org/latest/en/user/services/wms/reference.html
+          feature_count: 100,
+          x: point.x,
+          y: point.y
+        },
+        layerUrl,
+        true
+      )
+    );
+  }
+
+  redrawPopup(event) {
+    event.target.openPopup();
+    event.target.update();
+  }
+
+  showFeatureInfoMarkerAndFetchData (event) {
+    const latlng = event.latlng;
+    const wmsLayers = (this.props.tile.wmsLayers || []);
+    
+    // With long loading times it may be desirable to already show the popup with a spinner
+    this.setState({
+      featureInfo: {
+        show: true,
+        latlng: latlng,
+        data: null,
+      }
+    });
+    const mapRef = this.leafletMapRef.current.leafletElement;
+
+    // TODO make the showFeatureInfoMarkerAndFetchData also query raster and possibly background layer
+    // see commented out example code below
+    // on background no getFeatureInfoPossible ? because it is not wms
+    // const backgroundUrl = this.constructGetFeatureInfoUrl(
+    //   {
+    //     url: this.props.mapBackground.url,
+    //     name: this.props.mapBackground.description,
+    //     srs: "EPSG:4326",
+    //   },
+    //   mapRef,
+    //   latlng
+    // );
+    // do raster have getFeatureInfo ? -> didnot find tile with raster yet to test
```

Comment:
I think it should be possible to use a getFeatureInfo on a raster.

Examples I've encountered:

Parameter **format=image**
https://gis.stackexchange.com/questions/96113/json-issue-getfeatureinfo-with-raster-images

Or:

create a fake "vector" record out of
the raster band values and then proceed just like in the vector case
The band names become feature attributes, and their names depend on the type of raster
http://osgeo-org.1560.x6.nabble.com/getfeatureInfo-from-a-raster-layer-td5004891.html

Reply:
We should definitely look intoo this. But I might not be able before my holiday.
Joeri indeed would like to see the getFeatureInfo Popup also on raster.

Reply:
Anyway I think a first version does not require this feature. So we might merge and even go to production without featureInfo on rasters.

Reply:
I also agree on this.
But just in case, I also already wanted to point out that rasters should probably be possible, one way or the other.

Have a nice holiday! :)